src/main/java/org/opensearch/geospatial/index/mapper/xypoint/XYPointParser.java

```diff
@@ -0,0 +1,175 @@
+
+import java.io.IOException;
+import java.util.Collections;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.common.geo.GeoPoint;
+import org.opensearch.common.xcontent.LoggingDeprecationHandler;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.common.xcontent.XContentSubParser;
+import org.opensearch.common.xcontent.support.MapXContentParser;
+
+/**
+ * Parse the value and set XYPoint represented as a String, Object, WKT, array.
+ *
+ * @author Naveen Tatikonda
+ */
+public class XYPointParser {
+    public static final String X = "x";
+    public static final String Y = "y";
+    public static final String NULL_VALUE_PARAMETER = "null_value";
+
+    /**
+     * Parses the value and set XYPoint which was represented as an object.
+     *
+     * @param value  input which needs to be parsed which contains x and y coordinates in object form
+     * @param ignoreZValue  boolean parameter which decides if third coordinate needs to be ignored or not
+     * @param point  A {@link XYPoint} that will be reset by the values parsed
+     * @return {@link XYPoint} after setting the x and y coordinates parsed from the parse
+     * @throws OpenSearchParseException
+     */
+    public static XYPoint parseXYPoint(Object value, final boolean ignoreZValue, XYPoint point) throws OpenSearchParseException {
+        try (
+            XContentParser parser = new MapXContentParser(
+                NamedXContentRegistry.EMPTY,
+                LoggingDeprecationHandler.INSTANCE,
+                Collections.singletonMap(NULL_VALUE_PARAMETER, value),
+                null
+            )
+        ) {
+            parser.nextToken(); // start object
+            parser.nextToken(); // field name
+            parser.nextToken(); // field value
+            return parseXYPoint(parser, point, ignoreZValue);
+        } catch (IOException ex) {
+            throw new OpenSearchParseException("error parsing xy_point", ex);
+        }
+    }
+
+    /**
+     * Parse the values to set the XYPoint which was represented as a String, Object, WKT or an array.
+     *
+     * <ul>
+     *     <li> String: "100.35, -200.54" </li>
+     *     <li> Object: {"x" : 100.35, "y" : -200.54} </li>
+     *     <li> WKT: "POINT (-200.54 100.35)"</li>
+     *     <li> Array: [ -200.54, 100.35 ]</li>
+     * </ul>
+     *
+     * @param parser  {@link XContentParser} to parse the value from
+     * @param point  A {@link XYPoint} that will be reset by the values parsed
+     * @param ignoreZValue  boolean parameter which decides if third coordinate needs to be ignored or not
+     * @return {@link XYPoint} after setting the x and y coordinates parsed from the parse
+     * @throws IOException
+     * @throws OpenSearchParseException
+     */
+    public static XYPoint parseXYPoint(XContentParser parser, XYPoint point, final boolean ignoreZValue) throws IOException,
```

Comment:
what is the use of the function parameter : point? if you are returning the parsed object as point I don't think so you need to pass point as a parameter here.

Reply:
We actually don't need to pass XYPoint object as a parameter and we can instantiate it in XYPointParser. But, the [AbstractPointGeometryFieldMapper](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/mapper/AbstractPointGeometryFieldMapper.java#L304) class is passing it as a parameter to the parser in [XYPointFieldMapper](https://github.com/naveentatikonda/geospatial/blob/add_xy_point_field_mapper/src/main/java/org/opensearch/geospatial/index/mapper/xypoint/XYPointFieldMapper.java#L126) and if we don't pass it to this method in XYPointParser class it might create some confusion that why is this point being sent as a parameter to XYPointFieldMapper class

Reply:
Let's use the java doc and say that this parameter is of no use and make the XYPointParser interfaces clean.

Reply:
updated